### DISCUSSION TITLE
Request: programmatically customize report loc for typescript-eslint base rule extensions

### DISCUSSION THREAD

**Discussion by kirkwaiblinger** [eslint/eslint#18557] [General] [answered]
At typescript-eslint, we call into the rule objects from eslint rules for many of our "extension" rules, which add support for TS-specific syntax/behavior. 

Sometimes, we want to refine the report location coming from the base rule, since the JS AST node may include underlines on things that are not pertinent to the reason for the report (for example, type annotations).

In order to intercept the base rule's reports, we resort to a hacky proxy mechanism for the unproxiable `context` object in several places in our codebase. For example, see https://github.com/typescript-eslint/typescript-eslint/blob/3c6379b7678bcb190ae70d211cb3930c942d17a0/packages/eslint-plugin/src/rules/prefer-destructuring.ts#L184-L207

 Is there and/or can there be a more straightforward way to programmatically interact with the rule?
 
 Our main desired use cases would be 
 1. adjust rule's report `loc`, where the  AST node may include things not really relevant to the report (such as TS type annotations).
 2. Disable the fixer when the JS fix wouldn't be appropriate for TS code.
 
 See also discussion at https://github.com/typescript-eslint/typescript-eslint/pull/8893#discussion_r1623464316

**Comment by nzakas** [comment, marked as answer]
> Is there and/or can there be a more straightforward way to programmatically interact with the rule?

The short answer is no. This isn't a use case we formally support and ESLint was not designed for a plugin to pull in core rules and then mess with the output. What you're doing is incredibly fragile, and given that eventually there will be no core rules (everything will be in a plugin), we don't want to invest any time in more formally supporting what you're doing.

**Reply by kirkwaiblinger** [reply]
Understood. Thanks!
